### PR TITLE
fix: replace whole username for tab completion

### LINKED ISSUE #448: Chat mentions are case insensitive
A mention of `@User` incorrectly matches `@user`.

### PR DESCRIPTION
**Pull request by olastor**
should fix #448

### CODE REVIEW

File: src/app/message/message.component.ts [hunk 1]
```diff
@@ -59,13 +59,13 @@ export class MessageComponent implements OnInit, OnDestroy {
     // match usernames from chat
     const matches = this.usernames.filter(name => new RegExp('^' + str, 'i').test(name));
     if (matches.length > 0) {
-      this.currentMessage += matches[0].substring(str.length) + ' ';
+      this.currentMessage = this.currentMessage.slice(0, -1 * str.length) + matches[0];
```
Comment (binarious) [inline src/app/message/message.component.ts]: This does only work when the mentioning happens at the end of the text.

File: src/app/message/message.component.ts [hunk 2]
```diff
@@ -59,13 +59,13 @@ export class MessageComponent implements OnInit, OnDestroy {
...
     } else {
       this.userService.getUsername(str)
         .map(res => res.json())
         .subscribe(username => {
           if (username.length > 0) {
-            this.currentMessage += username.toString().substring(str.length) + ' ';
+            this.currentMessage = this.currentMessage.slice(0, -1 * str.length) + username;
```
Comment (binarious) [inline src/app/message/message.component.ts]: Same problem as above.

**Comment by binarious** [comment]
Can you fix the build, please?

**Comment by olastor** [comment]
please don't merge yet. Will rewrite it a bit.

**Comment by olastor** [comment]
Didn't work the way I wanted with re-setting the caret position after completing something within the text. But it should close the issue.

**Comment by binarious** [comment]
So it can be merged?

**Comment by olastor** [comment]
Yes

**Pull request merged**